main.go

```diff
@@ -132,63 +132,98 @@ func getProjectID(baseURL string, token string, projectname string, namespace st
 }
 
 // Get and return currently active milestones
-func getMilestones(baseURL string, token string, projectID string) (map[string]string, error) {
-	milestones := []milestoneAPI{}
-
+func getActiveMilestones(baseURL string, token string, projectID string) (map[string]string, error) {
 	strURL := []string{baseURL, projectID, "/milestones?state=active&per_page=100"}
 	URL := strings.Join(strURL, "")
+	milestones := map[string]string{}
 
-	m, linkHeader, err := callServer(URL, token, milestones)
+	m, err := gitlabGet(URL, token)
 	if err != nil {
-		return m, err
+		return milestones, err
 	}
 
-	for _, milestone := range milestones {
-		m[milestone.Title] = milestone.DueDate
+	for _, milestone := range m {
+		milestones[milestone.Title] = milestone.DueDate
 	}
 
-	parsedHeader := link.Parse(linkHeader)
-	for _, elem := range parsedHeader {
-		if elem.Rel == "next" {
-			m, _, err := callServer(URL, token, milestones)
-			if err != nil {
-				return m, err
-			}
-			for _, milestone := range milestones {
-				m[milestone.Title] = milestone.DueDate
-			}
-		}
+	return milestones, nil
+}
+
+// Get and return inactive milestones
+func getInactiveMilestones(baseURL string, token string, projectID string) (map[string]int, error) {
+	strURL := []string{baseURL, projectID, "/milestones?state=closed&per_page=100"}
+	URL := strings.Join(strURL, "")
+	milestones := map[string]int{}
+
+	m, err := gitlabGet(URL, token)
+	if err != nil {
+		return milestones, err
 	}
 
-	return m, nil
+	for _, milestone := range m {
+		milestones[milestone.Title] = milestone.ID
+	}
+
+	return milestones, nil
 }
 
-func callServer(URL string, token string, milestones []milestoneAPI) (map[string]string, string, error) {
-	m := map[string]string{}
+func gitlabGet(URL string, token string) ([]milestoneAPI, error) {
+	m := []milestoneAPI{}
 	client := &http.Client{}
-	var linkHeader string
+	
 	req, err := http.NewRequest("GET", URL, nil)
 	if err != nil {
-		return m, linkHeader, err
+		return m, err
 	}
-
+	
 	req.Header.Add("PRIVATE-TOKEN", token)
 	resp, err := client.Do(req)
 	if err != nil {
-		return m, linkHeader, err
+		return m, err
 	}
 
 	respByte, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return m, linkHeader, err
+		return m, err
 	}
 
-	json.Unmarshal(respByte, &milestones)
+	json.Unmarshal(respByte, &m)
 	defer resp.Body.Close()
-
+	var linkHeader string
 	linkHeader = resp.Header.Get("Link")
 
-	return m, linkHeader, nil
+	parsedHeader := link.Parse(linkHeader)
+	for _, elem := range parsedHeader {
```

Comment:
In my view this loop could be optimized down. We have quite similar code within the loop as above. That could be merged in one loop, but let's leave it for now and get the general structure worked out and working and then we can still optimize further.